gpt4all-backend/llmodel.cpp

```diff
@@ -51,10 +51,28 @@ Dlhandle *get_implementation(std::ifstream& f, const std::string& buildVariant)
 
 static
 bool requires_avxonly() {
-#ifdef __x86_64__
-    return !__builtin_cpu_supports("avx2") && !__builtin_cpu_supports("fma");
+#ifndef _MSC_VER
+    #ifdef __x86_64__
+        return !__builtin_cpu_supports("avx2") && !__builtin_cpu_supports("fma");
+    #else
+        #ifndef __WIN32__
+            #ifdef __aarch64__
+                return false;  // Don't know how to handle aarch64
+            #else
+                return false;  // Don't know how to handle other architectures
+            #endif
+        #else
+            return false;  // Windows doesn't support detection on architectures other than x86_64
+        #endif
+    #endif
 #else
-    return false;  // Don't know how to handle ARM
+    #ifdef __x86_64__
+        int cpuInfo[4];
+        __cpuidex(cpuInfo, 7, 0);
+        return !(cpuInfo[1] & (1 << 5));
+    #else
+        return false;  // Windows doesn't support detection on architectures other than x86_64
```

Comment:
This isn't specific to windows here what is this comment doing?

Reply:
It's specific for MSVC tho, and MSVC only compiles for Windows, right?